eclipse-collections/src/main/java/org/eclipse/collections/impl/list/primitive/LongInterval.java

```diff
@@ -0,0 +1,1092 @@
+        {
+            return false;
+        }
+        if (this.from == this.to)
+        {
+            return this.from == list.get(0);
+        }
+
+        if (otherList instanceof LongInterval)
+        {
+            LongInterval otherInterval = (LongInterval) otherList;
+            return (this.getFirst() == otherInterval.getFirst())
+                    && (this.getLast() == otherInterval.getLast())
+                    && (this.step == otherInterval.step);
+        }
+
+        if (this.from < this.to)
+        {
+            int listIndex = 0;
+            for (long i = this.from; i <= this.to; i += this.step)
+            {
+                if (i != list.get(listIndex++))
+                {
+                    return false;
+                }
+            }
+        }
+        else
+        {
+            int listIndex = 0;
+            for (long i = this.from; i >= this.to; i += this.step)
+            {
+                if (i != list.get(listIndex++))
+                {
+                    return false;
+                }
+            }
+        }
+        return true;
+    }
+
+    @Override
+    public int hashCode()
+    {
+        int hashCode = 1;
+        if (this.from == this.to)
+        {
+            hashCode = 31 * hashCode + (int) (this.from ^ this.from >>> 32);
+        }
+        else if (this.from < this.to)
+        {
+            for (long i = this.from; i <= this.to; i += this.step)
+            {
+                hashCode = 31 * hashCode + (int) (i ^ i >>> 32);
+            }
+        }
+        else
+        {
+            for (long i = this.from; i >= this.to; i += this.step)
+            {
+                hashCode = 31 * hashCode + (int) (i ^ i >>> 32);
+            }
+        }
+        return hashCode;
+    }
+
+    /**
+     * Returns a new LongInterval with the from and to values reversed and the step value negated.
+     */
+    @Override
+    public LongInterval toReversed()
+    {
+        return LongInterval.fromToBy(this.to, this.from, -this.step);
+    }
+
+    /**
+     * @since 6.0
+     */
+    @Override
+    public ImmutableLongList distinct()
+    {
+        return this;
+    }
+
+    @Override
+    public ImmutableLongList subList(int fromIndex, int toIndex)
+    {
+        throw new UnsupportedOperationException("subList not yet implemented!");
+    }
+
+    /**
+     * Returns the size of the interval.
+     */
+    @Override
+    public int size()
+    {
+        return this.size;
+    }
+
+    @Override
+    public long dotProduct(LongList list)
+    {
+        if (this.size() != list.size())
+        {
+            throw new IllegalArgumentException("Lists used in dotProduct must be the same size");
+        }
+        long sum = 0L;
+        for (int i = 0; i < this.size(); i++)
+        {
+            sum += this.get(i) * list.get(i);
+        }
+        return sum;
+    }
+
+    @Override
+    public boolean isEmpty()
+    {
+        return this.size() == 0;
```

Comment:
isn't this always false?

Reply:
Same issue in `IntInterval` (which I effectively cloned), can change implementation/add tests to both or have a separate PR to clean up this and other implementation issues. What do you suggest?

Reply:
For my understanding, is the proposal to return `false` instead of doing the check? Like Moh says it will always be `false` but this implementation is not incorrect or did I miss something? Interval always makes me think twice about these terminating methods because Interval is a mix of eager and lazy functionality.

Reply:
yes, the proposal is to return false. An implementation that computes PI to 700 digits and then returns `lastDigit == 0` would also work (the 700th digit of PI is 5) 😛 .

Reply:
LOL 😂